src/adagrams.js

```diff
@@ -1,15 +1,183 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+    const letterPool = {
+        A: 9,
+        B: 2,
+        C: 2,
+        D: 4,
+        E: 12,
+        F: 2,
+        G: 3,
+        H: 2,
+        I: 9,
+        J: 1,
+        K: 1,
+        L: 4,
+        M: 2,
+        N: 6,
+        O: 8,
+        P: 2,
+        Q: 1,
+        R: 6,
+        S: 4,
+        T: 6,
+        U: 4,
+        V: 2,
+        W: 2,
+        X: 1,
+        Y: 2,
+        Z: 1,
+    };
+    const letters = [
+        'A',
+        'B',
+        'C',
+        'D',
+        'E',
+        'F',
+        'G',
+        'H',
+        'I',
+        'J',
+        'K',
+        'L',
+        'M',
+        'N',
+        'O',
+        'P',
+        'Q',
+        'R',
+        'S',
+        'T',
+        'U',
+        'V',
+        'W',
+        'X',
+        'Y',
+        'Z',
+    ];
+    const randomLetters = (choices) =>
+        choices[Math.floor(Math.random() * choices.length)];
+    let hand = [];
+    while (hand.length != 10) {
+        let letter = randomLetters(letters);
+        for (const [key, value] of Object.entries(letterPool)) {
+            if (letter == key && value != 0) {
+                letterPool[key] -= 1;
+                hand.push(letter);
+            }
+        }
```

Comment:
This will make a full loop over the entries in `letterPool` each time we want to add a letter. Since we have a random letter to use as a key, I believe we could replace the for-loop with something like:
```
if (letterPool[letter] > 0) {
    letterPool[letter] -= 1;
    hand.push(letter)
}

```